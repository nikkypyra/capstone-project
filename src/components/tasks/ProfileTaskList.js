import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import TasksStyled from './TasksStyled'
import EditButton from '../buttons/EditButton'

ProfileTaskList.propTypes = {
  pet: PropTypes.object.isRequired,
  tasks: PropTypes.array.isRequired,
  handleCheckbox: PropTypes.func.isRequired,
  onClick: PropTypes.func,
}

export default function ProfileTaskList({
  pet,
  tasks,
  handleCheckbox,
  onClick,
}) {
  const todos = tasks.filter((task) => {
    return task.petId === pet.id
  })
  return (
    <>
      {todos
        .slice()
        .sort(
          (taskA, taskB) =>
            taskA.date.localeCompare(taskB.date) ||
            taskA.time.localeCompare(taskB.time)
        )
        .map((todo) => (
          <TaskWrapper key={todo.id}>
            <TasksStyled todo={todo} handleCheckbox={handleCheckbox} />
            <Update>
              <Link
                to={`/pet/${pet.id}/${todo.id}/update-task`}
                data-cy="edit-task"
              >
                <EditButton onClick={onClick} />
              </Link>
            </Update>
          </TaskWrapper>
        ))}
    </>
  )
}

const TaskWrapper = styled.section`
  margin: 16px 0px;
  display: grid;
  grid-template-columns: 1.4fr 2fr 3fr 1fr 1fr 1fr;
  grid-template-rows: auto;
  justify-content: space-evenly;
  align-items: center;
`
const Update = styled.div`
  grid-row: 1/2;
  grid-column: 6/7;
`
