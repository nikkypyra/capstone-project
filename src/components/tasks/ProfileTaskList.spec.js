import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import ProfileTaskList from './ProfileTaskList'

afterEach(cleanup)

const petMock = { name: 'Fluffy', id: '1' }
const tasksMock = [
  {
    id: '1',
    petId: '1',
    description: '',
    date: '',
    time: '',
    person: '',
    complete: true,
  },
]
const handleCheckbox = jest.fn()
const onClick = jest.fn()

describe('<ProfileTaskList />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ProfileTaskList
          pet={petMock}
          tasks={tasksMock}
          handleCheckbox={handleCheckbox}
          onClick={onClick}
        />
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
