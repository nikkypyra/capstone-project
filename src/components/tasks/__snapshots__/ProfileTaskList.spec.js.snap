// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileTaskList /> should render correctly 1`] = `
<DocumentFragment>
  <section
    class="ProfileTaskList__TaskWrapper-sc-1n7pogt-0 fFwAB"
  >
    <div
      class="TasksStyled__Marker-j2i96a-0 kTppzS"
    >
      <img
        alt=""
        src="taskpaw.png"
      />
    </div>
    <div
      class="TasksStyled__Description-j2i96a-1 dAkghC"
    >
      <h3 />
    </div>
    <div
      class="TasksStyled__Time-j2i96a-2 DoAZa"
    >
      <h4 />
    </div>
    <div
      class="TasksStyled__Date-j2i96a-3 kHFcCQ"
      data-cy="date"
    >
      <p />
    </div>
    <div
      class="TasksStyled__Person-j2i96a-4 jToTCz"
      data-cy="owner_name"
    >
      <p>
        To be completed by: 
      </p>
    </div>
    <div
      class="TasksStyled__Status-j2i96a-5 fCOCIY"
    >
      <input
        checked=""
        class="Checkbox__CheckboxStyled-sc-1blqffb-0 gefyIW"
        type="checkbox"
      />
    </div>
    <div
      class="ProfileTaskList__Update-sc-1n7pogt-1 hhSVAE"
    >
      <a
        data-cy="edit-task"
        href="/pet/1/1/update-task"
      >
        <div
          class="sc-AxjAm TegUP"
        >
          <img
            alt="edit"
            src="edit.png"
          />
        </div>
      </a>
    </div>
  </section>
</DocumentFragment>
`;
